src/e2ee.rs

```diff
@@ -151,38 +150,32 @@ pub fn try_decrypt(
     }
 
     /* possibly perform decryption */
-    let mut private_keyring = Keyring::default();
-    let mut public_keyring_for_validate = Keyring::default();
-    let mut out_mail = None;
+    let private_keyring: Keyring<SignedSecretKey> = Keyring::new_self(context)?;
```

Comment:
`new_self` guarantees this keyring has the private key loaded, simplifying the flow control below (and correcting their failure modes)  The rest of the logic below has not changed, it just drops a few levels of indentation.